From sebastian@breakpoint.cc  Wed Aug  6 16:48:04 2008
From: Sebastian Siewior <sebastian@breakpoint.cc>
Date: Thu, 17 Jul 2008 20:09:28 +0200
Subject: usb: return error code instead of 0 in the enqueue function.
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: linux-usb@vger.kernel.org, Sebastian Siewior <sebastian@breakpoint.cc>
Message-ID: <1216318170-23126-1-git-send-email-sebastian@breakpoint.cc>


if the enqueue function returns -ESHUTDOWN or -ENOMEM then
we return 0 instead of an error. This leads to a timeout and
then to a dequeue request of an not enqueued urb.

Signed-off-by: Sebastian Siewior <sebastian@breakpoint.cc>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/isp1760-hcd.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/usb/host/isp1760-hcd.c
+++ b/drivers/usb/host/isp1760-hcd.c
@@ -1615,8 +1615,7 @@ static int isp1760_urb_enqueue(struct us
 		return -EPIPE;
 	}
 
-	isp1760_prepare_enqueue(priv, urb, &qtd_list, mem_flags, pe);
-	return 0;
+	return isp1760_prepare_enqueue(priv, urb, &qtd_list, mem_flags, pe);
 }
 
 static int isp1760_urb_dequeue(struct usb_hcd *hcd, struct urb *urb,
